server/src/test/java/org/elasticsearch/index/snapshots/blobstore/SlicedInputStreamTests.java

```diff
@@ -79,6 +82,42 @@ protected InputStream openSlice(int slice) throws IOException {
         }
     }
 
+    public void testSkip() throws IOException {
+        final int slices = randomIntBetween(1, 20);
+        final var bytes = randomByteArrayOfLength(randomIntBetween(1000, 10000));
+        final int sliceSize = bytes.length / slices;
+
+        final var streamsOpened = new ArrayList<CheckClosedInputStream>();
+        SlicedInputStream input = new SlicedInputStream(slices) {
+            @Override
+            protected InputStream openSlice(int slice) throws IOException {
+                final int sliceOffset = slice * sliceSize;
+                final int length = slice == slices - 1 ? bytes.length - sliceOffset : sliceSize;
+                final var stream = new CheckClosedInputStream(new ByteArrayInputStream(bytes, sliceOffset, length));
+                streamsOpened.add(stream);
+                return stream;
+            }
+        };
+
+        // Skip up to a random point
+        final int skip = randomIntBetween(0, bytes.length);
+        input.skip(skip);
```

Comment:
We're relying on this skip always skipping the full amount here - we should at least assert that to be the case.

Reply:
Changed all skips in the test to skipNBytes as all of them are expected to skip the requested amount of bytes.